src/Lean/Elab/PreDefinition/Structural.lean

```diff
@@ -107,6 +109,7 @@ structure State where
      We use this information to generate the auxiliary `_sunfold` definition needed by the smart unfolding
      technique used at WHNF. -/
   matcherBelowDep : NameSet := {}
+  addMatchers : Array $ MetaM Unit := #[]
```

Comment:
`$` is not necessary here.

Reply:
yes it is...:
```lean
application type mismatch
  Array MetaM
argument
  MetaM
has type
  Type → Type : Type 1
but is expected to have type
  Type ?u.5843 : Type (?u.5843 + 1)
```

Unless you want `Array (MetaM Unit)`?

Reply:
I'm just about to push the update. Do you want `Array (MetaM Unit)` or `Array $ MetaM Unit`?